sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/degrade/circuitbreaker/CircuitBreaker.java

```diff
@@ -46,13 +50,12 @@
     State currentState();
 
     /**
-     * Record a completed request with the given response time and error (if present) and
-     * handle state transformation of the circuit breaker.
+     * Called when a passed invocation finished.
      *
-     * @param rt the response time of this entry
-     * @param error the error of this entry (if present)
+     * @param context context of current invocation
+     * @param wrapper current resource
      */
-    void onRequestComplete(long rt, Throwable error);
+    void afterRequestPassed(Context context, ResourceWrapper wrapper);
```

Comment:
Actually IMHO "complete" might be more concrete. In Sentinel, "pass" means allowed by Sentinel (not started but pending), and "complete" means an invocation has completed (the blocked requests never started, so never completed).

Reply:
And the `ResourceWrapper` seems unused? (it could be taken from `Entry` in `Context`)

Reply:
> pending

Yeah naming issues often have more than one answer mostly. I tried to make it clear for requests passed by sentinel(`passed`) and it will be invoked as it completed(`after`). Surely we can have other picks. Is `onRequestComplete` implicit whether sentinel guarantees it would be called no matter whether it was blocked?

Talk to the parameters in definition it just follows similar designs in related logics and try to make it more stable in future iterations. `ResourceWrapper` will contains the resource information which it's useful especially in implementations in future. My opinion is more like to keep it as stability of interface.

Reply:
> pending

Yeah naming issues often have more than one answer mostly. I tried to make it clear for requests passed by sentinel(`passed`) and it will be invoked as it completed(`after`). Surely we can have other picks. Is `onRequestComplete` implicit whether sentinel guarantees it would be called no matter whether it was blocked?

Talk to the parameters in definition it just follows similar designs in related logics and try to make it more stable in future iterations. `ResourceWrapper` will contains the resource information which it's useful especially in implementations in future. My opinion is more like to keep it as stability of interface.

Reply:
> Yeah naming issues often have more than one answer mostly. I tried to make it clear for requests passed by sentinel(`passed`) and it will be invoked as it completed(`after`). Surely we can have other picks. Is `onRequestComplete` implicit whether sentinel guarantees it would be called no matter whether it was blocked?

Originally the semantics of `onComplete` means the invocation has completed, so it MUST have passed Sentinel's checking. The rejected invocations never passed, so it would never complete.

> Talk to the parameters in definition it just follows similar designs in related logics and try to make it more stable in future iterations. `ResourceWrapper` will contains the resource information which it's useful especially in implementations in future. My opinion is more like to keep it as stability of interface.

Okay for me. Actually I'd like the circuit breaker to be a more generic design (e.g. the context could be generic, and metadata could be retrieved from it).

Reply:
> Okay for me. Actually I'd like the circuit breaker to be a more generic design (e.g. the context could be generic, and metadata could be retrieved from it).

Naming of callback has been reverted and it has more comments on it.
Talk to the generic context the idea makes sense but it also may introduce more memory footprints. Maybe it can be decided in future.

Reply:
Yes we may discuss it later in a new issue.